### PR TITLE
Database tasks can SKIP_TEST_DATABASE with environment variable

### PR DESCRIPTION
**Pull request by jmks**
### Summary

Small fix for https://github.com/rails/rails/issues/27299
Specifically, implements the comment here: https://github.com/rails/rails/issues/27299#issuecomment-617880652

Introduces environment variable `SKIP_TEST_DATABASE`, that when present, will not create/drop/etc. the test database.

Example:

```
➜ SKIP_TEST_DATABASE=true bundle exec rake db:create
Created database 'db/development.sqlite3'

➜ bundle exec rake db:create
Created database 'db/development.sqlite3'
Created database 'db/test.sqlite3'
```

### Other Information

I added a CHANGELOG entry for `activerecord` but think this should be documented somewhere.

I've searched the guides but have not found a "rake tasks" place where it would makes sense. I'm open to suggestions.

### CODE REVIEW

File: activerecord/test/cases/tasks/database_tasks_test.rb
Comment (bobwhitelock) [inline activerecord/test/cases/tasks/database_tasks_test.rb:450]:
```suggestion
    def test_creates_development_database_without_test_database_when_skip_test_database
```

**Comment by simi** [comment]
I left alt. idea at https://github.com/rails/rails/issues/27299#issuecomment-618531126.

File: activerecord/test/cases/tasks/database_tasks_test.rb
Comment (bobwhitelock) [inline activerecord/test/cases/tasks/database_tasks_test.rb:453]: What were your thoughts on the approach suggested in https://github.com/rails/rails/issues/27299#issuecomment-618107586, so an extra environment variable doesn't need to be set, rather only the dev database will be used when  `RAILS_ENV=development` is explicitly set? This might be slightly simpler/more intuitive, though possibly this could be confusing as there is different behaviour when `RAILS_ENV` is explicitly vs implicitly set to `development`

Comment (rafaelfranca) [reply activerecord/test/cases/tasks/database_tasks_test.rb:453]: Please read the issue you linked completely. I explain why only setting `RAILS_ENV=development` is not enough.

File: activerecord/CHANGELOG.md
Comment (rafaelfranca) [inline activerecord/CHANGELOG.md:1]:
```suggestion
*   Add `SKIP_TEST_DATABASE` environment variable to disabling modifying the test database when `rails db:create` and `rails db:drop` are called.
```

**Review by rafaelfranca — CHANGES_REQUESTED** [review]

File: activerecord/test/cases/tasks/database_tasks_test.rb
Comment (bobwhitelock) [reply activerecord/test/cases/tasks/database_tasks_test.rb:453]: Where? I can't see a comment from you mentioning `RAILS_ENV` directly

Comment (rafaelfranca) [reply activerecord/test/cases/tasks/database_tasks_test.rb:453]: https://github.com/rails/rails/issues/27299#issuecomment-295536459
https://github.com/rails/rails/issues/27299#issuecomment-295536872

File: activerecord/test/cases/tasks/database_tasks_test.rb
Comment (jmks) [reply activerecord/test/cases/tasks/database_tasks_test.rb]: Thanks!

File: activerecord/test/cases/tasks/database_tasks_test.rb
Comment (bobwhitelock) [reply activerecord/test/cases/tasks/database_tasks_test.rb:453]: Thanks, the former doesn't mention `RAILS_ENV` being explicitly set, but the latter reason makes sense. I've just been following the issue from a while back and so had forgotten about a linked commit message from 3 years ago when I got notified about the issue again.

**Comment by jmks** [comment]
@rafaelfranca Accepted the changes and resolved conflicts. Do you need anything else?

**Comment by jmks** [comment]
@rafaelfranca Friendly bump

**Comment by bobwhitelock** [comment]
Is anything blocking this being merged? Happy to help unblock it if so

**Comment by jmks** [comment]
I do not think there's any blockers. The changes requested by @rafaelfranca were re: the CHANGELOG.

I'll try to ping other AR maintainers this week.

File: activerecord/CHANGELOG.md
Comment (KapilSachdev) [inline activerecord/CHANGELOG.md:1]:
```suggestion
*   Add `SKIP_TEST_DATABASE` environment variable to disable modifying the test database when `rails db:create` and `rails db:drop` are called.
```

Either that or `... to disable modification of the test` or `...variable for disabling modification of the test`

`to disabling modifying the test` does not sounds right.

**Comment by jmks** [comment]
@rafaelfranca @kamipo @eugeneius @eileencodes Could I get some help getting this merged?

**Review by eileencodes — CHANGES_REQUESTED** [review]
@jmks can you fix the conflicts and rebase your commits so there is only 1 with a message about what is changing and why?

**Comment by jmks** [comment]
@eileencodes Done! The commits were small so I just squashed them.

(Looking at the Buildkite, Rails `master` seems to be failing now though)

**Review by eileencodes — APPROVED** [review]
I don't think the test failure here is related to the changes. I ran it a couple of times on CI (something I can do as an admin) and it's now failing that it can't find the image. I'm gonna merge this since I don't think it's causing the postgres failures.

**Comment by JohnDBR** [comment]
This will be available in the next rails release?

**Comment by ioquatix** [comment]
Just because I'm getting pinged about this issue, I would recommend Rails adopt a policy of using environment variables prefixed with `RAILS_` where possible, in order to avoid clobbering other systems/frameworks/etc and have a clear chain of responsibility/ownership. Anyway, I'm 👍🏼 on this improvement.

**Pull request merged**
